datadome/resource_custom_rule.go

```diff
@@ -2,42 +2,72 @@ package datadome
 
 import (
 	"context"
+	"fmt"
 	"strconv"
+	"time"
 
 	dd "github.com/datadome/terraform-provider/datadome-client-go"
+	"github.com/hashicorp/go-cty/cty"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+// resourceCustomRule define the CRUD operations and the schema definition for DataDome custom rules.
 func resourceCustomRule() *schema.Resource {
 	return &schema.Resource{
 		CreateContext: resourceCustomRuleCreate,
 		ReadContext:   resourceCustomRuleRead,
 		UpdateContext: resourceCustomRuleUpdate,
 		DeleteContext: resourceCustomRuleDelete,
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
-				ForceNew: true,
 			},
-			"query": &schema.Schema{
+			"query": {
 				Type:     schema.TypeString,
 				Required: true,
 			},
-			"response": &schema.Schema{
+			"response": {
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateDiagFunc: func(v any, p cty.Path) diag.Diagnostics {
+					var diags diag.Diagnostics
+					value := v.(string)
+					if !(value == "whitelist" || value == "captcha" || value == "block") {
```

Comment:
From the [public doc](https://docs.datadome.co/reference/post_1-1-protection-custom-rules), it is
- allow
- captcha
- block
--> this should be updated

Reply:
I agree that on the API Reference, it is written "allow".

But if you create your custom rule from Terraform, you will see that it is set as "whitelist" anyway (and can be disturbing if you update your custom rule and don't know this behaviour).
It seems like that is only a "front fix" to respect the audit.

We have 2 options:
- keep "whitelist" in order to don't have changes if you run a `terraform plan`  
- ask Dashboard team to also update this to "allow" in the back

Reply:
Code and example.tf file updated